### PR TITLE
doc/start: update os-recommendations.rst

### PR DESCRIPTION
**Pull request by zdover23**
This PR updates the extremely out-of-date
os-recommendations.rst file.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by anthonyeleven — APPROVED** [review]

**Comment by dvanders** [comment]
@zdover23 let's bring this to the CLT

**Comment by zdover23** [comment]
> @zdover23 let's bring this to the CLT

Roger that, DvdS.

File: doc/start/os-recommendations.rst [hunk 1]
```diff
@@ -41,6 +41,24 @@ platforms.  Generally speaking, there is very little dependence on
...
 package (i.e., sysvinit, systemd).
 
+Quincy
+------
+
++----------+----------+--------------------+--------------+---------+------------+
+| Distro   | Release  | Code Name          | Kernel       | Notes   | Testing    |
++==========+==========+====================+==============+=========+============+
+| CentOS   | 8        | N/A                | linux-4.18   |         | B, I, C    |
++----------+----------+--------------------+--------------+---------+------------+
```
Comment (jdurgin) [inline doc/start/os-recommendations.rst:51]: @djgalloway can help with a full list of what we're building for quincy + pacific.

We're testing both releases on centos stream 8, rhel 8, and ubuntu 20.04

Comment (djgalloway) [reply doc/start/os-recommendations.rst:51]: Here's what we *build* for (not necessarily support)
```
Octopus (15.X.X): focal bionic centos7 centos8 buster
Pacific (16.X.X): focal bionic centos8 buster bullseye
Quincy (17.X.X): focal centos8 bullseye
```

**Review by epuertat — COMMENTED** [review]
@zdover23 what about compressing all this data into a single table?

| Release Name | Tag | CentOS | Ubuntu | OpenSUSE | Debian |
|-----------------------|--------|--------|--------|----------|--------|
| **Quincy** | `17.2.z`  | 8 ✔️     | 20.04 ✔️ | 15.3     | 10 📁    |
| **Pacific** | `16.2.z`   |    8 ✔️   |    20.04 ✔️   |  15.2        |   10  📁   |
| **Octopus** | `15.2.z` |   7 ✅ [^1]   <br> 8 ✔️    |  18.04      |   15.2       |   9 📁      |

✔️ - packages + comprehensive testing
✅ - packages + basic testing
📁  - packages only

[^1]: ``btrfs`` is no longer tested on this release. We recommend using ``bluestore``

**Comment by zdover23** [comment]
jenkins retest this

**Comment by zdover23** [comment]
jenkins test docs

**Comment by zdover23** [comment]
jenkins test docs

**Comment by zdover23** [comment]
jenkins test docs

File: doc/start/os-recommendations.rst [hunk 2]
```diff
@@ -41,6 +41,42 @@ platforms.  Generally speaking, there is very little dependence on
 specific distributions aside from the kernel and system initialization
 package (i.e., sysvinit, systemd).
 
++--------------+--------+--------+--------+----------+--------+
+| Release Name | Tag    | CentOS | Ubuntu | OpenSUSE | Debian |
++==============+========+========+========+==========+========+
+| Quincy       | 17.2.z | 8      | 20.04  | 15.3     | 10     |
++--------------+--------+--------+--------+----------+--------+
+| Pacific      | 8      | 8      | 20.04  | 15.2     | 10     |
```
Comment (jdurgin) [inline doc/start/os-recommendations.rst:49]: pacific and octopus tag columns should be 16.2.z and 15.2.z

File: doc/start/os-recommendations.rst [hunk 3]
```diff
@@ -41,6 +41,42 @@ platforms.  Generally speaking, there is very little dependence on
...
++--------------+--------+--------+--------+----------+--------+
+| Octopus      | 20.04  | 8      | 18.04  | 15.2     | 10     |
++--------------+--------+--------+--------+----------+--------+
```
Comment (jdurgin) [inline doc/start/os-recommendations.rst:52]: debian should be 11 for quincy, and 10 and 11 for pacific like david mentioned. For pacific and octopus we have ubuntu 18.04 and 20.04, and for just octopus, centos 7 and 8..

**Comment by zdover23** [comment]
jenkins test docs

**Comment by zdover23** [comment]
jenkins render docs

File: doc/start/os-recommendations.rst (hunk 3 above)
Comment (zdover23) [reply doc/start/os-recommendations.rst:52]: I have included this information in this table.

**Comment by zdover23** [comment]
> @zdover23 what about compressing all this data into a single table?
> Release Name 	Tag 	CentOS 	Ubuntu 	OpenSUSE 	Debian
> **Quincy** 	`17.2.z` 	8 heavy_check_mark 	20.04 heavy_check_mark 	15.3 	10 file_folder
> **Pacific** 	`16.2.z` 	8 heavy_check_mark 	20.04 heavy_check_mark 	15.2 	10 file_folder
> **Octopus** 	`15.2.z` 	7 white_check_mark [1](#user-content-fn-1-6095d01d129cb64b1d5033d7d8660a94)
> 8 heavy_check_mark 	18.04 	15.2 	9 file_folder
> 
> heavy_check_mark - packages + comprehensive testing white_check_mark - packages + basic testing file_folder - packages only
> ## Footnotes
> 
>     1. `btrfs` is no longer tested on this release. We recommend using `bluestore` [leftwards_arrow_with_hook](#user-content-fnref-1-6095d01d129cb64b1d5033d7d8660a94)

https://github.com/ceph/ceph/pull/45782 includes this information.

**Comment by anthonyeleven** [comment]
That PR has a comment suggesting removal of at least some of it too.  I suggest that in both, Btrfs and BlueStore should not be backticked.

**Pull request merged**